lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -2089,4 +2072,33 @@ protected function switchPersisterContext($offset, $limit)
 
         $this->currentPersisterContext = $this->limitsHandlingContext;
     }
+
+    /**
+     * Retrieves class identifiers types
+     *
+     * @param ClassMetadata $class
+     * @return array
+     */
+    protected function getClassIdentifiersTypes(ClassMetadata $class)
```

Comment:
Please make this `private` for new methods. Add a return type too

Reply:
As far as I can see, this was just moved out, right?

Reply:
I made it `protected` so that the `JoinedSubclassPersister` could make use of it too. I'm not sure if it was moved out, I added it so that it can be used repeatedly.

Reply:
Hmm, I'm not sure about using inheritance for this. It makes sense, as long as the component stays internal and documented as such.

Looking at what the logic actually does, this seems like something that could live on its own though.

Reply:
@guilhermeblanco if you have time, can you check this and see what would be best (considering ORM 3.x changes)?

Reply:
OK, @guilhermeblanco, let me know if you want me to change this bit!